### ISSUE TITLE
routeRequest() render multiple routes

### ISSUE THREAD

**Issue by jatubio** [skipperbent/simple-php-router#452]
Hello,

I have updated to last version, and now, Router->routeRequest() method use a 'foreach' loop instead old 'for' loop. On this loop, in the old versions, if they found one route match, they exit with the last line:

`return $route->renderRoute($this->request);`

But, in the new version of this method, he performs a check to exit the loop:

                    $output = $route->renderRoute($this->request, $this);
                    if ($output !== null) {
                        return $output;
                    }

And if renderRoute() don't return anything, go back to the loop and if found another router that match, render also that route.

Why this change? Is there a way to configure that only renders a route?

Thank you

**Comment by skipperbent** [comment]
Hi.

You can stop the router from executing any other routes by returning the contents to it or using the exit function.

```php
class MyController {

     public function view() {
            return 'my-view-html';
     }

}

In this version you can basicly control when to stop the router. This is useful when creating large project like CMS-systems etc where multiple routes can be executed at once.

If you are still having issues with this, we could add an option to only render one route. Let me know if this is still an issue for you.

/ Simon

**Comment by jatubio** [comment]
Hi, thank you for reply :)

It was so much time ago that I can't barely remember what was my problem. I checked my code and I think that my problem was that my rendering route methods were rendering the content by themselves and don't returning anything.

So, I changed them to return `true`. Then, `SimpleRouter` was printing that `true` as `1` on my pages. So I override  `start()` method to return `''` because all my routes they render the content by themselves.

```
class Router extends SimpleRouter
{
    public function start(): ?string
    {
		parent::start();

		return '';
    }
}
```

Probably it's not a nice and good way to do it, but it's working and I'm not going to change that code anymore, hahaha :)

So, to me, the problem is solved, and it's up to you to decide if you want to consider this use case: Methods that handle the rendering themselves and, therefore return nothing. In that case, I think that to have a option to setup the router as 'only render one route' will simplify the things :)

**Comment by skipperbent** [comment]
Hi Jatubio.

Thanks for the detailed example and explanation, I really appreciate it. I've added an option for the router `SimpleRouter::enableMultiRouteRendering` where the multi-routing feature can be disabled (alias for `SimpleRouter::router()->setRenderMultipleRoutes($bool)`.

Unfortunately I can't set it as disabled per default as this will break peoples installation. This might be disabled by default in the next major release (5.0) as there seems to be a lot of confusion about this, which really indicate a bad design which is only used in advanced cases.

Thanks again.

/ Simon

**Comment by skipperbent** [comment]
This issue has been resolved in version 4.3.0.0.
Feel free to open another issue if you are still experiencing issues.
